pkg/odo/cli/registry/registry.go

```diff
@@ -0,0 +1,213 @@
+package registry
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/jedib0t/go-pretty/v6/table"
+	"github.com/jedib0t/go-pretty/v6/text"
+	"github.com/redhat-developer/odo/pkg/log"
+	"github.com/redhat-developer/odo/pkg/odo/cmdline"
+	"github.com/redhat-developer/odo/pkg/odo/genericclioptions"
+	"github.com/redhat-developer/odo/pkg/odo/genericclioptions/clientset"
+	odoutil "github.com/redhat-developer/odo/pkg/odo/util"
+	"github.com/redhat-developer/odo/pkg/registry"
+	"github.com/redhat-developer/odo/pkg/util"
+	"github.com/spf13/cobra"
+)
+
+const RecommendedCommandName = "registry"
+
+var Example = `  # Get all devfile components
+  %[1]s
+
+# Filter by name
+%[1]s --filter nodejs
+
+# Show more details
+%[1]s --details`
+
+// ListOptions encapsulates the options for the odo registry command
+type ListOptions struct {
+	clientset *clientset.Clientset
+
+	// List of known devfiles
+	DevfileList registry.DevfileStackList
+
+	// Flags
+	filterFlag  string
+	devfileFlag string
+	detailsFlag bool
+}
+
+// NewListOptions creates a new ListOptions instance
+func NewListOptions() *ListOptions {
+	return &ListOptions{}
+}
+
+func (o *ListOptions) SetClientset(clientset *clientset.Clientset) {
+	o.clientset = clientset
+}
+
+// Complete completes ListOptions after they've been created
+func (o *ListOptions) Complete(cmdline cmdline.Cmdline, args []string) (err error) {
+
+	o.DevfileList, err = o.clientset.RegistryClient.ListDevfileStacks("")
+	if err != nil {
+		return err
+	}
+
+	if o.DevfileList.DevfileRegistries == nil {
+		log.Warning("Please run 'odo preference registry add <registry name> <registry URL>' to add registry for listing devfile components\n")
+	}
+
+	return nil
+}
+
+// Validate validates the ListOptions based on completed values
+func (o *ListOptions) Validate() error {
+	if len(o.DevfileList.Items) == 0 {
+		return fmt.Errorf("no deployable components found")
+	}
+	return nil
+}
+
+// Run contains the logic for the command associated with ListOptions
+func (o *ListOptions) Run(ctx context.Context) (err error) {
+	o.printDevfileList(o.DevfileList.Items)
+	return nil
+}
+
+func NewCmdRegistry(name, fullName string) *cobra.Command {
+	o := NewListOptions()
+
+	var listCmd = &cobra.Command{
+		Use:     name,
+		Short:   "List all components from the Devfile registry",
+		Long:    "List all components from the Devfile registry",
+		Example: fmt.Sprintf(Example, fullName),
+		Run: func(cmd *cobra.Command, args []string) {
+			genericclioptions.GenericRun(o, cmd, args)
+		},
+	}
+
+	clientset.Add(listCmd, clientset.REGISTRY)
+
+	// Flags
+	listCmd.Flags().StringVar(&o.filterFlag, "filter", "", "Filter based on the name of the component")
+	listCmd.Flags().StringVar(&o.devfileFlag, "devfile", "", "Only show components from the specific Devfile stack")
```

Comment:
Playing with this, I typed individual Devfile stack name like `python` and `python-django`, but that failed. Passed the name of the registry and it worked. So, IMO, this should be reworded to below.:
```suggestion
	listCmd.Flags().StringVar(&o.devfileFlag, "devfile", "", "Only show components from the specific Devfile registry")
```